cyclonedx/model/bom.py

```diff
@@ -416,16 +494,49 @@ def vulnerabilities(self) -> "SortedSet[Vulnerability]":
     def vulnerabilities(self, vulnerabilities: Iterable[Vulnerability]) -> None:
         self._vulnerabilities = SortedSet(vulnerabilities)
 
-    @property
+    @property  # type: ignore[misc]
+    @serializable.xml_attribute()
     def version(self) -> int:
         return self._version
 
     @version.setter
     def version(self, version: int) -> None:
         self._version = version
 
+    @property  # type: ignore[misc]
+    @serializable.view(SchemaVersion1Dot2)
+    @serializable.view(SchemaVersion1Dot3)
+    @serializable.view(SchemaVersion1Dot4)
+    @serializable.xml_array(serializable.XmlArraySerializationType.NESTED, 'dependency')
+    @serializable.xml_sequence(5)
+    def dependencies(self) -> "SortedSet[Dependency]":
+        return self._dependencies
+
+    @dependencies.setter
+    def dependencies(self, dependencies: Iterable[Dependency]) -> None:
+        self._dependencies = SortedSet(dependencies)
+
+    def register_dependency(self, target: Dependable, depends_on: Optional[Iterable[Dependable]] = None) -> None:
+        _d = next(filter(lambda _d: _d.ref == target.bom_ref, self.dependencies), None)
+
+        if _d and depends_on:
+            # Dependency Target already registered - but it might have new dependencies to add
+            _d.dependencies = _d.dependencies.union(  # type: ignore
+                set(map(lambda _d: Dependency(ref=_d.bom_ref), depends_on)) if depends_on else []
+            )
+        elif not _d:
+            # First time we are seeing this target as a Dependency
+            self._dependencies.add(Dependency(
+                ref=target.bom_ref,
+                dependencies=list(map(lambda _dep: Dependency(ref=_dep.bom_ref), depends_on)) if depends_on else []
```

Comment:
:question: why cast the `map()` result to `list`? The `Dependency()` allows `dependencies` to be a `Iterable` - so stop casting here. If needed, it shall traverse/cast internally ...

Reply:
Apparently rubbish - so removed 🤷